src/Microsoft.Health.TaskManagement/JobHosting.cs

```diff
@@ -59,21 +59,47 @@ public async Task ExecuteAsync(byte queueType, short runningJobCount, string wor
                         JobInfo nextJob = null;
                         if (_queueClient.IsInitialized())
                         {
-                            try
+                            // Add retry logic for dequeuing
+                            int retryCount = 0;
+                            const int maxRetries = 3;
+                            bool succeeded = false;
+
+                            while (!succeeded && retryCount < maxRetries && !cancellationTokenSource.Token.IsCancellationRequested)
                             {
-                                _logger.LogInformation("Dequeuing next job.");
+                                try
+                                {
+                                    _logger.LogInformation("Dequeuing next job. Attempt {RetryCount}/{MaxRetries}", retryCount + 1, maxRetries);
+
+                                    if (checkTimeoutJobStopwatch.Elapsed.TotalSeconds > 600)
+                                    {
+                                        checkTimeoutJobStopwatch.Restart();
+                                        nextJob = await _queueClient.DequeueAsync(queueType, workerName, JobHeartbeatTimeoutThresholdInSeconds, cancellationTokenSource.Token, null, true);
```

Comment:
Could we move the retry into the `SqlQueueClient.DequeueAsync` method? This class is datastore agnostic - we shouldn't be exposing anything "SQL" in here if we can help it.